lib/RAST_SDK/RAST_SDKImpl.pm

```diff
@@ -1479,6 +1476,14 @@ sub annotate_genomes
 	my $genomes = $params->{input_genomes};
 
 	my $obj_type;
+		
+	my $empty_input_msg = ("ERROR:Missing required inputs--must specify at least one genome \n".
+	                       "and/or a string of genome names separated by ';', '\n' or '|' (without quotes).\n");
+	Bio::KBase::Exceptions::ArgumentValidationError->throw(
+            error        => $empty_input_msg,
+            method_name  => 'annotate_genomes'
+        ) unless (ref $genomes eq 'ARRAY' && @$genomes ) && $params->{ genome_text };
```

Comment:
Should this be checking for genomes OR genome_text?

Reply:
I added () to make it `stupid and simple` to understand, plus added comment to explain it.